parser-typechecker/src/Unison/PatternMatchCoverage/Desugar.hs

```diff
@@ -27,12 +27,9 @@ desugarMatch ::
   -- | scrutinee variable
   v ->
   -- | match cases
-  [MatchCase loc (Term' vt v loc)] ->
+  NonEmpty (MatchCase loc (Term' vt v loc)) ->
   m (GrdTree (PmGrd vt v loc) loc)
-desugarMatch loc0 scrutineeType v0 cs0 =
-  traverse desugarClause cs0 >>= \case
-    [] -> pure $ Leaf loc0
```

Comment:
This is the bug, an empty set of match cases is not a Leaf, it is an empty tree.